src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/NUnit/NUnitIntegration.cs

```diff
@@ -0,0 +1,145 @@
+            Scope scope = tracer.StartActive("nunit.test");
+            Span span = scope.Span;
+
+            span.Type = SpanTypes.Test;
+            span.SetMetric(Tags.Analytics, 1.0d);
+            span.SetTraceSamplingPriority(SamplingPriority.AutoKeep);
+            span.ResourceName = $"{testSuite}.{testName}";
+            span.SetTag(TestTags.Suite, testSuite);
+            span.SetTag(TestTags.Name, testName);
+            span.SetTag(TestTags.Framework, testFramework);
+            span.SetTag(TestTags.Type, TestTags.TypeTest);
+            CIEnvironmentValues.DecorateSpan(span);
+
+            var framework = FrameworkDescription.Instance;
+
+            span.SetTag(CommonTags.RuntimeName, framework.Name);
+            span.SetTag(CommonTags.RuntimeOSArchitecture, framework.OSArchitecture);
+            span.SetTag(CommonTags.RuntimeOSPlatform, framework.OSPlatform);
+            span.SetTag(CommonTags.RuntimeProcessArchitecture, framework.ProcessArchitecture);
+            span.SetTag(CommonTags.RuntimeVersion, framework.ProductVersion);
+
+            // Get test parameters
+            ParameterInfo[] methodParameters = testMethod.GetParameters();
+            if (methodParameters?.Length > 0)
+            {
+                for (int i = 0; i < methodParameters.Length; i++)
+                {
+                    if (testMethodArguments != null && i < testMethodArguments.Length)
+                    {
+                        span.SetTag($"{TestTags.Arguments}.{methodParameters[i].Name}", testMethodArguments[i]?.ToString() ?? "(null)");
+                    }
+                    else
+                    {
+                        span.SetTag($"{TestTags.Arguments}.{methodParameters[i].Name}", "(default)");
+                    }
+                }
+            }
+
+            // Get traits
+            if (testMethodProperties != null)
+            {
+                skipReason = (string)testMethodProperties.Get("_SKIPREASON");
+                foreach (var key in testMethodProperties.Keys)
+                {
+                    if (key == "_SKIPREASON")
+                    {
+                        continue;
+                    }
+
+                    IList value = testMethodProperties[key];
+                    IEnumerable<string> values = Enumerable.Empty<string>();
+                    if (value != null)
+                    {
+                        List<string> lstValues = new List<string>();
+                        foreach (object valObj in value)
+                        {
+                            if (valObj is null)
+                            {
+                                continue;
+                            }
+
+                            lstValues.Add(valObj.ToString());
+                        }
+
+                        values = lstValues;
+                    }
+
+                    span.SetTag($"{TestTags.Traits}.{key}", string.Join(", ", values) ?? "(null)");
```

Comment:
nit: I know this is almost a direct copy of the previous code, but I just noticed something. The `?? "(null)"` expression in `string.Join(", ", values) ?? "(null)"` looks unnecessary and can be removed. Based on my reading of the code, `values` will never be null (though it may have zero elements). As a result, `string.Join` will always return a string with a real result or the empty string, thus the `??` operator will never be used.

Were you thinking of setting the value to `"(null)"` if the list was empty and `string.Join` returned an empty string?

Reply:
Oh yeah, thanks for the catch, you're right, I changed in https://github.com/DataDog/dd-trace-dotnet/pull/1131/commits/b07c228c42ae967b0a730a3498d9e20d55879659